frontend/src/metabase/common/components/EntityPicker/components/ResultItem/ResultItem.tsx

```diff
@@ -0,0 +1,91 @@
+import { t } from "ttag";
+
+import { Ellipsified } from "metabase/core/components/Ellipsified";
+import { color } from "metabase/lib/colors";
+import { getIcon } from "metabase/lib/icon";
+import { Icon, Flex, Tooltip } from "metabase/ui";
+import type { SearchResult } from "metabase-types/api";
+
+import { ChunkyListItem } from "./ResultItem.styled";
+
+export type ResultItemType = Pick<
+  SearchResult,
+  | "model"
+  | "collection"
+  | "name"
+  | "description"
+  | "collection_authority_level"
+  | "moderated_status"
+  | "display"
+>;
+
+export const ResultItem = ({
+  item,
+  onClick,
+  isSelected,
+  isLast,
+}: {
+  item: ResultItemType;
+  onClick: () => void;
+  isSelected?: boolean;
+  isLast?: boolean;
+}) => {
+  const icon = getIcon({
+    ...item,
+    authority_level:
+      item.model === "collection"
+        ? item?.collection_authority_level
+        : item.moderated_status,
+  });
+
+  return (
+    <ChunkyListItem onClick={onClick} isSelected={isSelected} isLast={isLast}>
+      <Flex gap="md" miw="10rem" align="center" style={{ flex: 1 }}>
+        <Icon
+          color="brand"
+          name={icon.name}
+          style={{
+            color: color(icon.color ?? (isSelected ? "white" : "brand")),
+            flexShrink: 0,
+          }}
+        />
+        <Ellipsified style={{ fontWeight: "bold" }}>{item.name}</Ellipsified>
+        {item.description && (
+          <Tooltip
+            maw="20rem"
+            multiline
+            label={item.description}
+            styles={{
+              tooltip: {
+                // required to get the tooltip to show up over the entity picker modal 😢
+                zIndex: "450 !important" as any,
```

Comment:
I feel very bad about this, but the default zIndex for tooltips is 300, and the entity picker modal is at z-index 400 so it doesn't get overlapped by notifications that should be underneath it. Open to alternative ideas!

Reply:
nit and proof that i'm a fuddy duddy, i've seen emojis break tooling in the past and prefer to avoid using them in code.

Reply:
my only real feedback about the use of a very specific z-index # is that the context gets lots. i've previously used a constants file in the past that records all z-indexes so folks can understand the hierarchy in one file.

Reply:
yeah, this is a result of a series of sins. I will 
1) remove descriptions from this PR altogether so that this can merge
2) in a separate PR, hopefully tomorrow, fix the series of z-index sins in a separate PR
3) open a third PR to put descriptions back into this branch

Reply:
Or can't you merge and just improve z-index sins in a follow up? That does sound like a lot only to avoid a magic number 🤔

Reply:
hooray! i haven't had an opportunity to post my favorite meme lately.
![backlog](https://github.com/metabase/metabase/assets/30528226/8e2023e5-06fb-4077-b775-d34ff9e6e93f)

Reply:
https://github.com/metabase/metabase/pull/42272